packages/analytics/src/factory.ts

```diff
@@ -44,11 +44,38 @@ import {
   areCookiesEnabled,
   isBrowserExtension
 } from '@firebase/util';
+import { initializeIds } from './initialize-ids';
+import { logger } from './logger';
+
+interface FirebaseAnalyticsInternal extends FirebaseAnalytics {
```

Comment:
You should implement this [interface](https://github.com/firebase/firebase-js-sdk/blob/master/packages/app-types/private.d.ts#L38).

Reply:
Changed it to extend both, since it is implemented as a POJO and not a class, is that ok?